synapse/models/doc.py

```diff
@@ -0,0 +1,93 @@
+import synapse.exc as s_exc
+import synapse.lib.module as s_module
+
+class DocModule(s_module.CoreModule):
+
+    def getModelDefs(self):
+        return (('doc', {
+            'interfaces': (
+                ('doc:document', {
+
+                    'doc': 'A common interface for documents.',
+
+                    'template': {
+                        'type': 'NEWP',
```

Comment:
Is this vestigial?

Reply:
not exactly. It makes it blow up with `NoSuchType` for NEWP rather than a missing format specifier. It should only ever be triggered at dev time when you've implemented the interface and overlooked specifying a type. Happy to tweak, but it seemed clearer.